src/main/java/org/javamodularity/moduleplugin/tasks/TestTask.java

```diff
@@ -56,20 +58,11 @@ public void configureTestJava(Project project, String moduleName) {
             TestEngine.select(project).ifPresent(testEngine -> {
                 args.addAll(List.of("--add-reads", moduleName + "=" + testEngine.moduleName));
 
-                Set<String> testPackages = new HashSet<>();
-                for (var sourceFile : testSourceSet.getJava()) {
-                    try {
-                        Optional<PackageDeclaration> optionalPackageDeclaration = JavaParser.parse(sourceFile).getPackageDeclaration();
-                        if (optionalPackageDeclaration.isPresent()) {
-                            PackageDeclaration packageDeclaration = optionalPackageDeclaration.get();
-                            testPackages.add(packageDeclaration.getNameAsString());
-                        }
-                    } catch (FileNotFoundException e) {
-                        throw new UncheckedIOException(e);
-                    }
+                PackageScanner scanner = new PackageScanner();
```

Comment:
I think it makes sense to change the public API of PackageScanner so that it accepts List of source files instead a single file. I do not see any scenario when only a single file needs to be parsed.

Reply:
I agree.